va/drm/va_drm_utils.c

```diff
@@ -117,15 +116,14 @@ VA_DRM_GetDriverName(VADriverContextP ctx, char **driver_name_ptr, int candidate
 int
 VA_DRM_IsRenderNodeFd(int fd)
 {
-    struct stat st;
     const char *name;
 
     /* Check by device node */
-    if (fstat(fd, &st) == 0)
-        return S_ISCHR(st.st_mode) && (st.st_rdev & 0x80);
+    if (drmGetNodeTypeFromFd(fd) == DRM_NODE_RENDER)
+        return 1;
 
     /* Check by device name */
-    name = drmGetDeviceNameFromFd(fd);
+    name = drmGetDeviceNameFromFd2(fd);
```

Comment:
This function call ultimately results into `strdup`. Hm, don't we have a memory leak here? /and had it before since v1 of this function used `strdup` as well/

https://github.com/freedesktop/mesa-drm/blob/10dd3eb6d5ab6c1867e1ca9dd67756c152e6269e/xf86drm.c#L2792
https://github.com/freedesktop/mesa-drm/blob/10dd3eb6d5ab6c1867e1ca9dd67756c152e6269e/xf86drm.c#L4295

Reply:
@jbeich : ping